cmd/kubeadm/app/util/system/docker_validator.go

```diff
@@ -51,18 +58,28 @@ func (d *DockerValidator) Validate(spec SysSpec) (error, error) {
 		return nil, nil
 	}
 
-	c, err := client.NewClient(dockerEndpoint, "", nil, nil)
+	// Run 'docker info' with a JSON output and unmarshal it into a dockerInfo object
+	info := dockerInfo{}
+	out, err := exec.Command("docker", "info", "--format", "{{json .}}").Output()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to create docker client")
+		return nil, errors.Wrap(err, `failed executing "docker info --format '{{json .}}'"`)
```

Comment:
Probably worth while to put the output in the error, maybe also use `CombinedOutput` to capture stderr above?

Reply:
yes, makes sense. i've added it.